### PR TITLE
[java] Test multisig quorum parameter conditions

### PR DESCRIPTION
**Pull request by andozw**
Adding tests for the multisig quorum parameter conditions.

### CODE REVIEW

**Comment by ivmaykov** [comment]
@andozw did you mean to merge into a non-master branch?

**Comment by andozw** [comment]
> @andozw did you mean to merge into a non-master branch?

Compare to non-master branch, yes. I created this branch off of the PR that I just merged.

File: java/.gitignore [hunk 1]
```diff
@@ -1,4 +1,9 @@
-# Note: this file is a copy of https://github.com/github/gitignore/blob/main/Gradle.gitignore
+# Ignore gradlew files
+gradle/wrapper/
+gradlew
+gradlew.bat
+
+# Note: this rest of the file is a copy of https://github.com/github/gitignore/blob/main/Gradle.gitignore
```
Comment (ivmaykov) [inline java/.gitignore:6]: I thought this was already done in #711 ? Why is it showing up here as well?

File: java/shared/src/test/java/com/squareup/subzero/shared/SubzeroUtilsTest.java [hunk 2]
```diff
@@ -139,6 +189,38 @@ public void setUp() {
...
+    assertEquals("index should be between 0 and 2^31-1", exception.getMessage());
+  }
+
+  @Test public void testDerivePublicKeyMaxIndex() {
+    DeterministicKey rootKey = HDKeyDerivation.createMasterPrivateKey(new byte[32]);
+    // Integer.MAX_VALUE is 2^31-1.
+    Path path = Path.newBuilder().setIsChange(false).setIndex(Integer.MAX_VALUE).build();
+
+    DeterministicKey derivedKey = SubzeroUtils.derivePublicKey(rootKey, path);
+    assertEquals(33, derivedKey.getPubKey().length);
```
Comment (ivmaykov) [inline java/shared/src/test/java/com/squareup/subzero/shared/SubzeroUtilsTest.java:209]: If we have a hard-coded root key (looks like we're using an all-zeroes key here?), can we hard-code the expected derived key here and check for equality?

**Review by ivmaykov — APPROVED** [review]
LGTM with some questions/comments

File: java/.gitignore (hunk 1 above)
Comment (andozw) [reply java/.gitignore:6]: github updated the base branch automatically but I think I had to make a change to the earlier commits so I will need to rebase this branch

**Review by ivmaykov — APPROVED** [review]
Merge it!

**Pull request merged**